src/hooks/useSnackbar/useSnackbar.tsx

```diff
@@ -1,76 +1,86 @@
-import { snackbarTransitions } from '@/shared/components/Snackbar'
-import Snackbar from '@/shared/components/Snackbar/Snackbar'
-import { transitions } from '@/shared/theme'
-import { Global } from '@emotion/react'
-import React, { createContext, useContext, useEffect, useState } from 'react'
-import { CSSTransition } from 'react-transition-group'
+import React, { createContext, useContext, useState, useRef, useEffect } from 'react'
+import { CSSTransition, TransitionGroup } from 'react-transition-group'
+import styled from '@emotion/styled'
+import { Snackbar } from '@/shared/components'
+import { transitions, sizes } from '@/shared/theme'
+import { createId } from '@/utils/createId'
 
 export type DisplaySnackbarArgs = {
-  time?: number
-  variant?: 'success' | 'error' | 'info'
-  buttonText?: string
-  message: string
+  timeout?: number
+  variant?: 'primary' | 'secondary'
+  icon?: 'success' | 'error' | 'info'
+  title: string
+  description?: string
+  actionText?: string
+  onActionClick?: () => void
 }
 
+type SnackbarsState = {
+  id: string
+} & Omit<DisplaySnackbarArgs, 'time'>
+
 type SnackbarContextValue = {
   displaySnackbar: (args: DisplaySnackbarArgs) => void
-  closeSnackbar: () => void
+  closeSnackbar: (id: string) => void
```

Comment:
There actually isn't any way for the consumer to know this ID, right?

Reply:
Let's return it from `displaySnackbar`

Reply:
Done